### PR TITLE
Added comments for generic timer.

### PR DESCRIPTION
**Pull request by HyeJi**
Added comments on those file list.

./hypervisor/generic_timer/generic_timer.h
./hypervisor/generic_timer/generic_timer.c

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

**Comment by Wonseok** [comment]
Please remember this: good title begins with a verb with past present because it tells us what you did. You might want to fix the title, e.g. Added comments for generic timer. Your change do not cover every timer in our hypervisor such as pwm timer or sp804 timer.

File: hypervisor/generic_timer/generic_timer.c [hunk 1]
```diff
@@ -130,6 +130,16 @@ static inline uint64_t generic_timer_reg_read64(int reg)
     return val;
 }
 
+/** @brief Registers generic timer irqs such as hypervisor timer event
+ *  (GENERIC_TIMER_HYP), non-secure physical timer event(GENERIC_TIMER_NSP)
+ *  and virtual timer event(GENERIC_TIMER_NSP).
+ *  Each interrup source is identified by a unique ID.
```
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:136]: You might want to add information more, how could I get the interrupt sources of each timer?
Please add page and name of the part in arm technical reference manual.

File: hypervisor/generic_timer/generic_timer.c [hunk 2]
```diff
@@ -139,6 +149,9 @@ hvmm_status_t generic_timer_init()
     return HVMM_STATUS_SUCCESS;
 }
 
+/** @brief Configures time interval by PL2 physical timerValue register.
+ *  (VMSA : CNTHP_TVAL) Convets measurement from microseconds to count.
```
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:153]: s/Convets/Convert

File: hypervisor/generic_timer/generic_timer.c [hunk 3]
```diff
@@ -139,6 +149,9 @@ hvmm_status_t generic_timer_init()
...
+ */
```
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:154]: Could you add the information how to convert microseconds to count?

File: hypervisor/generic_timer/generic_timer.c (hunk 2 above)
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:153]: You might want to use period(.) after "(VMSA : CNTHP_TVAL)". e.g. (VMSA : CNTHP_TVAL).

File: hypervisor/generic_timer/generic_timer.c [hunk 4]
```diff
@@ -152,6 +165,10 @@ hvmm_status_t generic_timer_set_tval(uint32_t tval)
     return result;
 }
 
+/** @brief Enables the timer interrupt such as hypervisor timer event
+ *  by PL2 Physical Timer Control register(VMSA : CNTHP_CTL)
```
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:169]: Add period.

File: hypervisor/generic_timer/generic_timer.c [hunk 5]
```diff
@@ -189,6 +209,8 @@ static void _generic_timer_hyp_irq_handler(int irq, void *regs, void *pdata)
     _callback[GENERIC_TIMER_HYP](regs);
 }
 
+/** @brief Enables irq.
```
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:212]: What the IRQ enable in this function?

File: hypervisor/generic_timer/generic_timer.c [hunk 6]
```diff
@@ -206,6 +228,9 @@ hvmm_status_t generic_timer_enable_irq(void)
     return result;
 }
 
+/** @brief Configures callback by add callback.
+ *  Called when occur timer interrupt.
```
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:232]: What does it mean?
Registers timer callback for each timer such as XXX, XXX.

Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:232]: Each timer callback are registered with the timer interrupt source.
A timer callback called when the timer interrupt occurs.

File: hypervisor/generic_timer/generic_timer.c [hunk 7]
```diff
@@ -215,6 +240,8 @@ hvmm_status_t generic_timer_set_callback(timer_callback_t callback,
     return HVMM_STATUS_SUCCESS;
 }
 
+/** @brief tries dump.
```
Comment (Wonseok) [inline hypervisor/generic_timer/generic_timer.c:243]: What does 'tries' mean?

**Pull request closed without merging**